service/auth.go

```diff
@@ -48,6 +58,38 @@ func (s *Service) authHandler(w http.ResponseWriter, r *http.Request) (clientID
 	return clientID, http.StatusOK, nil
 }
 
+func (s *Service) bearerAuthHandler(w http.ResponseWriter, r *http.Request) (clientID string, code int, err error) {
+	bearerToken, ok := bearerAuth(r)
+	if !ok {
+		return "", http.StatusUnauthorized, fmt.Errorf("authentication failed: invalid auth header")
+	}
+
+	session, err := s.sessionCache.Get(bearerToken)
+	if err != nil {
+		return "", http.StatusUnauthorized, fmt.Errorf("authentication failed: %w", err)
+	}
+	clientID = session.ClientID
+
+	if clientID == "" {
+		return "", http.StatusUnauthorized, fmt.Errorf("authentication failed: unauthorized")
+	}
+
+	return clientID, http.StatusOK, nil
+}
+
+func bearerAuth(r *http.Request) (token string, ok bool) {
+	auth := r.Header.Get("Authorization")
+	return parseBearerAuth(auth)
+}
+
+func parseBearerAuth(auth string) (token string, ok bool) {
+	// Case-insensitive prefix match
+	if len(auth) < len(bearerPrefix) || !strings.EqualFold(strings.ToLower(auth[:len(bearerPrefix)]), strings.ToLower(bearerPrefix)) {
```

Comment:
Why do we lowercase the strings? I'd expect `EqualFold` to take care of that already.